### PR TITLE
Issue#49: Replace Guzzle Http Client with Symfony Client

### PR DESCRIPTION
**Pull request by mpash**
This PR tackles issue #49 
Please see the altered test changes for `getReasonPhrase`, otherwise all tests pass like before with the new client.

### CODE REVIEW

File: Consul/Tests/Services/KVTest.php [hunk 1]
```diff
@@ -82,7 +80,7 @@ public function testDeleteWithDefaultOptions()
             $this->fail('fail because the key does not exist anymore.');
         } catch (\Exception $e) {
             $this->assertInstanceOf('SensioLabs\Consul\Exception\ClientException', $e);
-            $this->assertContains('404 - Not Found', $e->getMessage());
```
Comment (mpash) [inline Consul/Tests/Services/KVTest.php:85]: Symfony has no equivalent for `getReasonPhrase` so I've removed this functionality here in the test and in the client.

**Comment by mpash** [comment]
The failed builds are to be expected, with adopting `symfony/http-client` the minimum PHP version [has been moved to v7.2 or greater](https://github.com/symfony/http-client/blob/5.x/composer.json).

**Comment by nicolas-grekas** [comment]
7.1 should work fine with http-client 4.4:
https://github.com/symfony/http-client/blob/4.4/composer.json

**Comment by mpash** [comment]
@nicolas-grekas I absolutely will not update to an unsupported version of PHP.
image
https://www.php.net/supported-versions.php

**Review by lyrixx — APPROVED** [review]
Nice

Thanks @mpash For this PR.

**Comment by lyrixx** [comment]
Released a V4.0.0 https://github.com/FriendsOfPHP/consul-php-sdk/releases/tag/v4.0.0

**Pull request merged**